Tools/CISupport/ews-build/steps.py

```diff
@@ -2150,6 +2174,188 @@ def hideStepIf(self, results, step):
+            if len(label_as_blocked):
+                self.build.addStepsAfterCurrentStep([SafeMergeToMergeBlocked()])
+
+        defer.returnValue(SUCCESS if num_prs else FAILURE)
+
+    def getResultSummary(self):
+        if self.results == SUCCESS:
+            return {'step': f"Successfully requested and filtered pull requests"}
+        elif self.results == FAILURE:
+            return {'step': f"Failed to request and filter pull requests"}
+        return buildstep.BuildStep.getResultSummary(self)
+
+    @defer.inlineCallbacks
+    def getNumPRs(self, label='\"safe-merge-queue\"'):
+        query_body = '{ repository(owner:"Webkit", name:"WebKit") { pullRequests(labels: %s) { totalCount } } }' % label
+        query = {'query': query_body}
+        rc = yield self.query_graph_ql(query)
+        if rc:
+            num_prs = rc['data']['repository']['pullRequests']['totalCount']
+            yield self._addToLog('stdio', 'There are {} PR(s) in safe-merge-queue.\n'.format(num_prs))
+            defer.returnValue(num_prs)
+        else:
+            yield self._addToLog('stdio', 'Failed to retrieve number of PRs.\n')
+            defer.returnValue(None)
+
+    @defer.inlineCallbacks
+    def sortPRs(self, num_prs, label='\"safe-merge-queue\"'):
+        query_body = '{ repository(owner:"Webkit", name:"WebKit") { pullRequests(labels: %s, last: %s) { edges { node { title number commits(last: 3) { nodes { commit { commitUrl status { state contexts { context state } } } } } } } } } }' % (label, num_prs)
+        query = {'query': query_body}
+
+        yield self._addToLog('stdio', "Fetching all PRs with label {}...\n".format(label))
+
+        response = yield self.query_graph_ql(query)
+        if not response:
+            yield self._addToLog('stderr', 'Failed to retrieve list of PRs.\n')
+            defer.returnValue(None)
+
+        all_pr_data = response['data']['repository']['pullRequests']['edges']
+        list_of_prs = []
+        passed_status_check = []
+        failed_status_check = []
+        no_checks = []
+        ios_tvos_watchos = ['ios', 'mac', 'ios-sim', 'ios-wk2', 'ios-wk2-wpt', 'api-ios', 'tv', 'tv-sim', 'watch', 'watch-sim']
+        macos = ['mac', 'mac-AS-debug', 'api-mac', 'mac-wk1', 'mac-wk2', 'mac-AS-debug-wk2', 'mac-wk2-stress']
+        check_tests = ios_tvos_watchos + macos
+
+        for pr_data in all_pr_data:
+            failed = False
+            pr_num = pr_data['node']['number']
+            pr_commit_status_data = pr_data['node']['commits']['nodes'][0]['commit']['status']
+
+            list_of_prs.append(pr_num)
+
+            yield self._addToLog('stdio', 'Checking the status of PR {}...\n'.format(pr_num))
+            if pr_commit_status_data is None:
+                no_checks.append(pr_num)
+                yield self._addToLog('stdio', 'No status. Done!\n')
+            else:
+                pr_commit_state = pr_commit_status_data['state']
```

Comment:
You shouldn't retrieve EWS data about a commit from GitHub checks. We might stop sending that data there (e.g.: https://github.com/WebKit/WebKit/pull/4767).

Instead, get the data from ews.webkit.org, that service has holistic view of the PR/commit status and has detailed information about the PR/commit. You can use/modify `CheckStatusOnEWSQueues` step.